README.md

```diff
@@ -1,22 +1,59 @@
 # Elasticsearch
 
-This repository contains Ruby integrations for [Elasticsearch](https://www.elastic.co/products/elasticsearch):
+This repository contains Ruby integrations for [Elasticsearch](https://www.elastic.co/products/elasticsearch).
 
-* a client for connecting to an Elasticsearch cluster,
-* a Ruby API for the Elasticsearch's REST API,
-* a Ruby API for the X-Pack APIs,
-* various extensions and utilities.
+The [`elasticsearch`](https://github.com/elasticsearch/elasticsearch-ruby/tree/master/elasticsearch)
+library is a wrapper for two separate libraries:
+
+* [`elasticsearch-transport`](https://github.com/elasticsearch/elasticsearch-ruby/tree/master/elasticsearch-transport),
+  which provides a low-level Ruby client for connecting to an [Elasticsearch](https://www.elastic.co/products/elasticsearch) cluster
+* [`elasticsearch-api`](https://github.com/elasticsearch/elasticsearch-ruby/tree/master/elasticsearch-api),
+  which provides a Ruby API for the Elasticsearch RESTful API
+
+```ruby
+require 'elasticsearch'
+
+client = Elasticsearch::Client.new log: true
+
+# if you specify Elasticsearch host
+# client = Elasticsearch::Client.new url: 'http://localhost:9200', log: true
+
+client.transport.reload_connections!
+
+client.cluster.health
+
+client.search q: 'test'
+
+# etc.
+```
+
+Both of these libraries are extensively documented.
+**Please read the [`elasticsearch-transport`](http://rubydoc.info/gems/elasticsearch-transport) and the [`elasticsearch-api`](http://rubydoc.info/gems/elasticsearch-api) documentation carefully.**
+
+See also [`doc/examples`](https://github.com/elastic/elasticsearch-ruby/blob/master/docs/examples/README.md) for some practical examples.
+
+_Keep in mind, that for optimal performance, you should use a HTTP library which supports persistent
+("keep-alive") connections, e.g. [Patron](https://github.com/toland/patron) or [Typhoeus](https://github.com/typhoeus/typhoeus)._ These libraries are not dependencies of the elasticsearch gems, so be sure to define a dependency in your own application.
+
+This repository contains these additional Ruby libraries:
+
+* [`elasticsearch-extensions`](https://github.com/elastic/elasticsearch-ruby/tree/master/elasticsearch-extensions),
+   which provides a set of extensions to the base library,
+* [`elasticsearch-dsl`](https://github.com/elastic/elasticsearch-ruby/tree/master/elasticsearch-dsl),
+  which provides a Ruby API for the [Elasticsearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html),
+* [`elasticsearch-xpack`](https://github.com/elastic/elasticsearch-ruby/tree/master/elasticsearch-xpack),
+  which provides Ruby API for the  [_X-Pack_](https://www.elastic.co/products/x-pack) APIs. This API is going to be merged into `elasticsearch-api` on v8.0.
```

Comment:
`https://www.elastic.co/products/x-pack` redirects to `https://www.elastic.co/elastic-stack` now.
I don't think we need the italics or link here anymore as this is soon going away.

```suggestion
  which provides Ruby API for X-Pack APIs. This API is going to be merged into `elasticsearch-api` on v8.0.
```